packages/js-commons/src/core/local-files.ts

```diff
@@ -184,6 +184,24 @@ export const getLocalApplicationTransporter = (
       await setGlobalMetadata(appsMetaData);
       return appData;
     },
+
+    delete: async (appId: string): Promise<void> => {
+      const appsMetaData = await getGlobalMetadata();
+      const existingAppMetadata = appsMetaData[appId];
+      if (!existingAppMetadata) {
+        throw Error(`App ${appId} not found on local disk`);
+      }
+      // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+      delete appsMetaData[appId];
```

Comment:
This should happen only after the directory is removed. Otherwise if something goes wrong with deleting the directory you will have lost the pointer.